ebos/eclgenericproblem_impl.hh

```diff
@@ -168,14 +169,15 @@ readRockParameters_(const std::vector<Scalar>& cellCenterDepths,
         rockTableIdx_.resize(numElem);
         const auto& num = eclState_.fieldProps().get_int(rock_config.rocknum_property());
         for (std::size_t elemIdx = 0; elemIdx < numElem; ++ elemIdx) {
-            rockTableIdx_[elemIdx] = num[elemIdx] - 1;
+            rockTableIdx_[elemIdx] = num[this->getLookUpData(elemIdx)] - 1;
             auto fmtError =
-                [&num,elemIdx,&ijkIndex,&rock_config](const char* type, std::size_t size)
+                [&num,elemIdx,&ijkIndex,&rock_config, this](const char* type, std::size_t size)
                 {
```

Comment:
Would it be cleaner to introduce a variable with name e.g. `coarseElemIdx`, which holds the result of the lookup? Then that could be captured instead of this in the lambda.